File: tarea 2/tarea2.java
/*import java.awt.Color;
import java.awt.BasicStroke;
import javax.swing.JFrame;
import javax.swing.JPanel;
import java.awt.Graphics;
import java.awt.Graphics2D;
import java.awt.geom.Ellipse2D;

public class Punto {
    public float x;
    public float y;

    public Punto(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float[] coord_cartesianas() {
        return new float[]{x, y};
    }

    public float[] coord_polares() {
        float r = (float) Math.sqrt(x * x + y * y);
        float theta = (float) Math.atan2(y, x);
        return new float[]{r, theta};
    }

    @Override
    public String toString() {
        return "Punto{" + "x=" + x + ", y=" + y + '}';
    }
}
//LINEA
class Linea extends JPanel {

    private Punto punto1;
    private Punto punto2;

    public Linea(Punto punto1, Punto punto2) {
        this.punto1 = punto1;
        this.punto2 = punto2;
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        Graphics2D g2 = (Graphics2D) g;
        g2.setColor(Color.BLUE);
        g2.setStroke(new BasicStroke(2));

        g2.draw(new Line2D.Double(punto1.x * 100, punto1.y * 100, punto2.x * 100, punto2.y * 100));
    }

    public static void main(String[] args) {
        Punto punto1 = new Punto(3, 1);
        Punto punto2 = new Punto(1, 5);

        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.add(new Linea(punto1, punto2));
        frame.setSize(400, 400);
        frame.setVisible(true);

//------------------------------------------------------
//CIRCUNFERENCIA

public class Circunferencia extends JPanel {

    private Punto centro;
    private float radio;

    public Circunferencia(Punto centro, float radio) {
        this.centro = centro;
        this.radio = radio;
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        Graphics2D g2 = (Graphics2D) g;
        g2.setColor(Color.RED);
        g2.setStroke(new BasicStroke(2));

        double x = centro.x * 100 - radio * 100;
        double y = centro.y * 100 - radio * 100;
        double width = radio * 200;
        double height = radio * 200;

        g2.draw(new Ellipse2D.Double(x, y, width, height));
    }

    public static void main(String[] args) {
        Punto centro = new Punto(2, 2);
        float radio = 1.5f; 

        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.add(new Circunferencia(centro, radio));
        frame.setSize(400, 400);
        frame.setVisible(true);
    }
}

File: tarea 2/tarea2.py
import math
import numpy as np
import matplotlib.pyplot as plt

class Punto:
    def __init__(self, x: float, y: float):
        self.x = x
        self.y = y
    def coord_cartesianas(self):
        return [self.x, self.y]
    def coord_polares(self):
        r = math.sqrt(self.x ** 2 + self.y ** 2)
        theta = math.atan2(self.y, self.x)
        return [r, theta]
    def __str__(self):
        return f'Punto(x={self.x}, y={self.y})'
punto1 = Punto(3, 1)
punto2 = Punto(1, 5)

#LINEA
x_coords = [punto1.x, punto2.x]
y_coords = [punto1.y, punto2.y]

plt.plot(x_coords, y_coords, label="Línea entre puntos", marker="o")
plt.axhline(0, color='black', linewidth=0.5)
plt.axvline(0, color='black', linewidth=0.5)
plt.grid(color='gray', linestyle='--', linewidth=0.5)
plt.legend()
plt.gca().set_aspect('equal', adjustable='box')
plt.title('Línea en base a la Clase Punto')
plt.xlabel('X')
plt.ylabel('Y')

plt.show()

#----------------------------------------------------------------------

#CIRCUNFERENCIA

centro = Punto(2, 2)
radio = float(1.5)
theta = np.linspace(0, 2 * np.pi, 100)
x_circle = centro.x + radio * np.cos(theta)
y_circle = centro.y + radio * np.sin(theta)

plt.plot(x_circle, y_circle, label=f'Circunferencia (Centro: {centro}, Radio: {radio})')
plt.axhline(0, color='black', linewidth=0.5)
plt.axvline(0, color='black', linewidth=0.5)
plt.grid(color='gray', linestyle='--', linewidth=0.5)
plt.legend()
plt.gca().set_aspect('equal', adjustable='box')
plt.title('Circunferencia en base a la Clase Punto')
plt.xlabel('X')
plt.ylabel('Y')

plt.show()
